### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by owenhelloworld**
- Add dev docker files 
- Add docker compose
- Add instructions to run local Mongo container, so you don't need to run the Mongo service
- Fixed some build issues
- Added default route that can be used as health check in the future

Possible future changes:
- Mount local machine dev files so containers don't need to be rebuilt. This will require a separate set of production grade containers to be built and pushed to hub with CI/CD

### CODE REVIEW

**Review by jameslindfors — COMMENTED** [review]
These are some suggestions. I'll let @minhngo3818 make the final review.

**Comment by minhngo3818** [comment]
I can't provide sufficient feedback for this PR because I'm unfamiliar with Docker, and probably most of DevOps tools at the moment. Look like James does the review well, so I think he can handle this PR.

File: backend/index.js [hunk 1]
```diff
@@ -37,7 +35,9 @@ app.use(express.json())
 // const organizationsRoute = require("./routes/organizations/orgs-route")
 
 // Use Routes
-
+app.get("/", (req, res) => {
+    res.send("GET request to the homepage")
+})
```
[the remark this answers is not in the document]
Comment (owenhelloworld) [reply backend/index.js:40]: / should have something either wayso its not just a 5xx or 4xx. Future FR for health one day IMO to check other dependancies ?

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      dockerfile: Dockerfile.dev
+   ports:
+    -  3000:3000
+   container_name: occ-client
+   networks:
+    -  occ-net
+   depends_on:
+    - backend
+
+  backend:
```
[the remark this answers is not in the document]
Comment (owenhelloworld) [reply docker-compose.yaml:16]: Operational environment should do health checks, not containers themselves.

File: README.md [hunk 3]
```diff
@@ -5,3 +5,31 @@ https://computer-science-club-occ.github.io/
...
+Rebuild All Containers:
+
+`docker-compose build`
+
+### Cleaning Docker Env
+**Deleting the mongo container will delete all data in the database**
+1. Stop the container(s) using the following command:
+`docker-compose down`
+2. Delete all containers using the following command:
+`docker rm -f $(docker ps -a -q)`
```
[the remark this answers is not in the document]
Comment (owenhelloworld) [reply README.md]: sorry old bad habit of mine to do it the old fashioned way :) . Committed.

**Review by jameslindfors — APPROVED** [review]
Good PR. Containerizing the app was a good idea.

**Pull request merged**